uproot/write/objects/TH.py

```diff
@@ -48,43 +48,33 @@ def __init__(self, histogram):
             raise NotImplementedError(self.fClassName)
         elif self.fClassName == b"TH2C":
-            self.valuesarray = numpy.array(self.values, dtype=">i1")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">i1").transpose()
         elif self.fClassName == b"TH2S":
-            self.valuesarray = numpy.array(self.values, dtype=">i2")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">i2").transpose()
         elif self.fClassName == b"TH2I":
-            self.valuesarray = numpy.array(self.values, dtype=">i4")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">i4").transpose()
         elif self.fClassName == b"TH2F":
-            self.valuesarray = numpy.array(self.values, dtype=">f4")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">f4").transpose()
         elif self.fClassName == b"TH2D":
-            self.valuesarray = numpy.array(self.values, dtype=">f8")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">f8").transpose()
         elif self.fClassName == b"TProfile2D":
             raise NotImplementedError(self.fClassName)
         elif self.fClassName == b"TH3C":
-            self.valuesarray = numpy.array(self.values, dtype=">i1")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">i1").transpose()
         elif self.fClassName == b"TH3S":
-            self.valuesarray = numpy.array(self.values, dtype=">i2")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">i2").transpose()
         elif self.fClassName == b"TH3I":
-            self.valuesarray = numpy.array(self.values, dtype=">i4")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">i4").transpose()
         elif self.fClassName == b"TH3F":
-            self.valuesarray = numpy.array(self.values, dtype=">f4")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">f4").transpose()
         elif self.fClassName == b"TH3D":
-            self.valuesarray = numpy.array(self.values, dtype=">f8")
-            raise NotImplementedError(self.fClassName)
+            self.valuesarray = numpy.array(self.values, dtype=">f8").transpose()
         elif self.fClassName == b"TProfile3D":
             raise NotImplementedError(self.fClassName)
         else:
             raise ValueError("unrecognized histogram class name {0}".format(self.fClassName))
 
-        self.fields["_fNcells"] = len(self.valuesarray)
+        self.fields["_fNcells"] = self.valuesarray.size
```

Comment:
I prefer using `len(array)` rather than `array.size`, but I wouldn't make a big deal about it.

Reply:
Using `len(array)` instead of `array.size` would break the code for TH2 and TH3.

Reply:
Yes, sorry about not thinking this through before shoving stuff @reikdas. I'm fixing this now.

Reply:
@reikdas Okay, I have now unreverted the change. I am going to just leave this alone until tomorrow when there has been more time to debug the test suite.

I ran the test suit on my Linux machine and everything passed fine, so I'm confused as to what it going on here. :? Sorry for causing extra annoyances here.